### PR TITLE
Fix level cutscenes sometimes not filling the screen

### LINKED ISSUE #365: [Runtime Issue]: Sometimes level 5’s cutscene doesn’t fill the screen
### Build Version

v1.5.0 56E61275

### Operating System Environment

- [ ] Microsoft Windows (32-bit)
- [ ] Microsoft Windows (64-bit)
- [ ] Mac OS X
- [X] Linux (specify distribution and version below)

### CPU Environment

- [ ] x86 (32-bit Intel/AMD)
- [X] x86\_64 (64-bit Intel/AMD)
- [ ] ARM (32-bit)
- [ ] ARM64 (64-bit; sometimes called AArch64)
- [ ] Other (RISC V, PPC...)

### Game Modes Affected

- [X] Single player
- [ ] Anarchy
- [ ] Hyper-Anarchy
- [ ] Robo-Anarchy
- [ ] Team Anarchy
- [ ] Capture the Flag
- [ ] Bounty
- [ ] Entropy
- [ ] Hoard
- [ ] Monsterball
- [ ] Cooperative

### Game Environment

I’m running NixOS 23.11 and using game data from the Windows Dual-Jewel version of Descent 3 v1.4.

### Description

If you start a new game on Descent 3: Retribution level 5, then the intro cutscene will fill the entire screen. However, if you start on a different level and then make it to level 5, the intro cutscene will only fill the top left corner of the screen.

### Regression Status

I don’t know.

### Steps to Reproduce

1. Run the game with a resolution that’s larger than the title screen’s resolution:
    ```
    ./Descent3 --width 1920 --height 1080 --aspect 1.7777778
    ```
3. Start a new game on any level of Descent 3: Retribution.
4. Type in the MORECLANG cheat code.
5. Warp to level 5.

### PR DESCRIPTION
**Pull request by pzychotic**
## Pull Request Type

- [ ] GitHub Workflow changes
- [ ] Documentation or Wiki changes
- [ ] Build and Dependency changes
- [x] Runtime changes
  - [ ] Render changes
  - [ ] Audio changes
  - [ ] Input changes
  - [ ] Network changes
  - [x] Other changes

### Description

Calling `SetScreenMode(SM_CINEMATIC)` right before starting the level cutscenes, so they use the correct resolution to render.

### Related Issues

Fixes #365

### Checklist

- [x] I have tested my changes locally and verified that they work as intended.
- [x] I have documented any new or modified functionality.
- [x] I have reviewed the changes to ensure they do not introduce any unnecessary complexity or duplicate code.
- [x] I understand that by submitting this pull request, I am agreeing to license my contributions under the project's license.

### CODE REVIEW

File: Descent3/gamesequence.cpp [hunk 1]
```diff
@@ -1401,6 +1401,7 @@ bool DoLevelIntro() {
 #endif
   //	if this level has a movie:
   if (lvl->flags & LVLFLAG_STARTMOVIE) {
+    SetScreenMode(SM_CINEMATIC);
```
Comment (winterheart) [inline Descent3/gamesequence.cpp:1404]: Probably we need this for `LVLFLAG_ENDMOVIE` too?

Comment (pzychotic) [reply Descent3/gamesequence.cpp:1404]: Indeed, good catch!
Glad we have <DEL+E> to end level instantly. Tested on Retribution Level 15 where this happened as well.
Moving the `SetScreenMode()` call into `DoMissionMovie()` to fix it.

**Comment by winterheart** [comment]
Please resolve conflict and we will be ready to merge.

**Comment by pzychotic** [comment]
> Please resolve conflict and we will be ready to merge.

Done.

**Pull request merged**